py/serial_keel.py

```diff
@@ -362,13 +362,22 @@ async def write(self, endpoint: Endpoint, message: str):
         assert response == {'Sync': 'WriteOk'}
 
     def endpoint_messages(self, endpoint: Endpoint) -> EndpointMessages:
+        '''
+        Retrieves  messages received by the endpoint.
```

Comment:
Decided to add this so that we don't forget. At least with this you get the idea that you don't need to try-catch...if you don't want to.

Reply:
Hmm, I don't think this does what is intended.
This just returns `EndpointMessages`, so it's not awaitable, and will therefore never raise a timeout.

It's the `async def __anext__(self)` in `EndpointMessages` that would need to have a `try .. except` I think.
So if that class also had a field which stated the name of the endpoint we could log that if the exception occurs there.